pkgs/by-name/st/styluslabs-write/package.nix

```
@@ -0,0 +1,99 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenv,
+  gitMinimal,
+  pkg-config,
+  SDL2,
+  xorg,
+  dbus,
+  libcap,
+  libGL,
+  roboto,
+  imagemagick,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "styluslabs-write";
+  version = "2024-09-26";
+
+  src = fetchFromGitHub {
+    owner = "styluslabs";
+    repo = "Write";
+    rev = "7110e5db87c154d28efd9507be4ab768af7403e0";
+    hash = "sha256-/IYI/trC14IBYtYn9LkjNrMZoCn6jtQIgq0ZqnsLISs=";
+    fetchSubmodules = true;
+  };
+
+  patches = [ ./sdl2_link_dynamic.patch ];
+
+  preBuild = ''
+    pushd syncscribble
+    # Bundled with a SDL2 fork to which it is statically linked.
+    # For nixpkgs, we apply a patch to dynamically link to SDL2 instead.
+    # TODO: We may still need to use this fork for non-Linux platforms.
+    rm -rf SDL/
+  '';
+
+  postBuild = ''
+    popd
+  '';
+
+  strictDeps = true;
+
+  hardeningDisable = [ "format" ];
+  makeFlags = [
+    "DEBUG=0"
+    "GITREV=${finalAttrs.src.rev}"
+  ];
+
+  nativeBuildInputs = [
+    imagemagick # magick
+    pkg-config
+    gitMinimal
+    SDL2 # sdl2-config
+  ];
+
+  buildInputs = [
+    SDL2
+    xorg.libX11
+    xorg.libXi
+    xorg.libXcursor
+    xorg.libXfixes
+    libGL
+    dbus.lib
+    libcap.lib
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,opt}
+    install -m555 -D syncscribble/Release/Write $out/opt/
+    install -m444 -D scribbleres/Intro.svg $out/opt/
+    ln -s ${roboto}/share/fonts/truetype/Roboto-Regular.ttf $out/opt/Roboto-Regular.ttf
+
+    ln -s ../opt/Write $out/bin/Write
+
+    for i in 16 24 48 64 96 128 256 512; do
+      mkdir -p $out/share/icons/hicolor/''${i}x''${i}/apps
+      magick scribbleres/write_512.png -resize ''${i}x''${i} $out/share/icons/hicolor/''${i}x''${i}/apps/${finalAttrs.pname}.png
+    done
+
+    install -Dm444 scribbleres/linux/Write.desktop -t $out/share/applications
+    substituteInPlace $out/share/applications/Write.desktop \
+        --replace-fail 'Exec=/opt/Write/Write' 'Exec=Write' \
+        --replace-fail 'Icon=Write144x144' 'Icon=${finalAttrs.pname}'
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = {
+    homepage = "https://styluslabs.com/";
+    description = "Cross-platform (Windows, Mac, Linux, iOS, Android) application for handwritten notes.";
+    license = lib.licenses.agpl3Only;
+    maintainers = with lib.maintainers; [ lukts30 ];
```

I'm familiar enough with the drv by now and know the program well enough to review.

If you'd like, feel free to add me to the maintainers :) 

```suggestion
    maintainers = with lib.maintainers; [ lukts30 atemu ];
```